normandy/schema-check/test_api.py

```diff
@@ -0,0 +1,285 @@
+import configparser
+import pytest
+import requests
+
+
+@pytest.fixture
+def conf():
+    config = configparser.ConfigParser()
+    config.read('manifest.ini')
+    return config
+
+
+def test_recipes(conf, env):
+    r = requests.get(conf.get(env, 'api_root') + '/recipe')
+    response = r.json()
+    expected_fields = [
+        'id',
+        'last_updated',
+        'name',
+        'enabled',
+        'revision_id',
+        'action',
+        'arguments',
+        'filter_expression',
+        'current_approval_request',
+        'approval',
+        'is_approved'
+    ]
+
+    expected_arguments = [
+        'defaults',
+        'surveyId',
+        'surveys'
+    ]
+
+    expected_survey_fields = [
+        'message',
+        'thanksMessage',
+        'engagementButtonLabel',
+        'title',
+        'postAnswerUrl',
+        'weight',
+        'learnMoreUrl',
+        'learnMoreMessage',
+    ]
+
+    expected_default_fields = [
+        'message',
+        'thanksMessage',
+        'engagementButtonLabel',
+        'postAnswerUrl',
+        'learnMoreUrl',
+        'learnMoreMessage'
+    ]
+
+    # Verify we have at least one response
+    assert len(response) >= 1
+
+    # Take a look at the first response and look for the expected fields
+    record = response[0]
+    for field in expected_fields:
+        assert field in record
+
+    # Do the arguments look right?
+    for argument in expected_arguments:
+        assert argument in record['arguments']
+
+    # Do the defaults look right?
+    for field in record['arguments']['defaults']:
+        assert field in expected_default_fields
+
+    # Do survey fields look right? Look at the first one
+    for field in record['arguments']['surveys'][0]:
+        assert field in expected_survey_fields
+
+
+def filter_show_heartbeat(record):
+    return record['name'] == 'show-heartbeat'
+
+
+def test_actions(conf, env):
+    r = requests.get(conf.get(env, 'api_root') + '/action')
+    response = r.json()
+
+    # Verify we have at least one response and then grab the first record
+    assert len(response) >= 1
+    record = response[0]
+
+    # Does an 'action' have all the required fields?
+    expected_action_fields = [
+        'name',
+        'implementation_url',
+        'arguments_schema'
+    ]
+    for field in record:
+        assert field in expected_action_fields
+
+    # Do we have the correct 'arguments_schema' fields?
+    expected_arguments_schema_fields = [
```

Comment:
Similar to my comment above, the only hard requirement of the `arguments_schema` field is that it is a valid JSON Schema; not having a `description` field would still be a valid response from the API.

The [jsonschema](https://github.com/Julian/jsonschema/) library has some functions that can help with validating a schema if you're interested in that.